src/throttledPromiseAll.ts

```diff
@@ -18,6 +18,15 @@ export type PromiseItem<T, O = T | undefined> = {
   producer: (source: T, throttledPromise: ThrottledPromiseAll<T, O | undefined>) => Promise<O | undefined>;
 };
 
+type IndexedProducer<T, O = T> = PromiseItem<T, O> & {
```

Comment:
Internal use type to add index to input and output.

Index is used to maintain result order to align results with Promise.all